src/frontends/pytorch/src/op/im2col.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_im2col(NodeContext& context) {
+    auto input = context.get_input(0);
+    auto kernel_size = context.const_input<std::vector<int64_t>>(1);
+    FRONT_END_OP_CONVERSION_CHECK(kernel_size.size() == 2, "kernel size should contains 2 elements");
+    auto dilation = context.const_input<std::vector<int64_t>>(2);
+    FRONT_END_OP_CONVERSION_CHECK(kernel_size.size() == 2, "dilation should contains 2 elements");
+    auto padding = context.const_input<std::vector<int64_t>>(3);
+    FRONT_END_OP_CONVERSION_CHECK(kernel_size.size() == 2, "padding should contains 2 elements");
+    auto stride = context.const_input<std::vector<int64_t>>(4);
+    FRONT_END_OP_CONVERSION_CHECK(kernel_size.size() == 2, "stride should contains 2 elements");
+    auto input_shape = context.mark_node(std::make_shared<opset8::ShapeOf>(input));
+    auto zero = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {0}));
+    auto zero_f = context.mark_node(opset8::Constant::create(element::f32, Shape{}, {0}));
+    auto minus_one = context.mark_node(opset8::Constant::create(element::i64, Shape{1}, {-1}));
+    auto two = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {2}));
+    auto four = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {4}));
+    auto b_dim = zero;
+    auto c_dim = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {1}));
+    auto h_dim = two;
+    auto w_dim = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {3}));
+    auto input_c = context.mark_node(std::make_shared<opset8::Gather>(input_shape, c_dim, zero));
```

Comment:
Instead of `Gather` you can do this using `Split` it will be simpler.

Reply:
looks like it does not work inside translator and brings more difficulties, or maybe I do something wrong :)
```c++
 auto input_shape = context.mark_node(std::make_shared<opset8::ShapeOf>(input));
 auto input_shape_split = context.mark_node(std::make_shared<opset8::Split>(input_shape, const_zero, 4)
 auto input_b = input_shape_split ->output(0).get_shared_node_ptr();
 auto input_c = input_shape_split ->output(1).get_shared_node_ptr();
 auto input_h = input_shape_split ->output(2).get_shared_node_ptr();
 auto input_w = input_shape_split ->output(3).get_shared_node_ptr();
```
tests failed because all of input_ variable are equal of input_b

Reply:
do not do get_shared_node_ptr()

Reply:
updated, thanks